subsys/bluetooth/enocean.c

```diff
@@ -217,6 +220,26 @@ static void handle_switch_data(const struct bt_le_scan_recv_info *info,
 	cb->button(dev, action, status >> 1, opt_data, opt_data_len);
 }
 
+static void handle_switch_payload(const struct bt_le_scan_recv_info *info,
+				  struct net_buf_simple *buf, const uint8_t *payload, uint8_t len,
+				  bool has_shortened_name)
+{
+	/* The data format is decided by a mix of lengths and bitfields */
+	if (len == 29 || (has_shortened_name && len == 23)) {
+		if (commissioning) {
+			handle_switch_commissioning(info, buf, has_shortened_name);
+		}
+		return;
```

Comment:
Nitpick: Could the commissioning flag check be moved inside `handle_switch_commissioning`? I feel it might make it a bit cleaner